Release/tests/functional/http/listener/listener_construction_tests.cpp

```diff
@@ -549,9 +549,12 @@ XzJTD4slrGSJrcpLt/g/Jqqdjg==
         listener.open().wait();
 
         client::http_client_config client_config;
+#if !defined(_WIN32) && !defined(__cplusplus_winrt) || defined(CPPREST_FORCE_HTTP_CLIENT_ASIO)
         client_config.set_ssl_context_callback(
             [&](boost::asio::ssl::context& ctx) { ctx.add_certificate_authority(cert); });
-
+#else
+        client_config.set_validate_certificates(false);
```

Comment:
Looking at this again I don't think we can do this -- it is better to fail to build than to product an implementation failing to meet basic HTTPS guarantees, which include validating the certificate. Unless there's something I'm missing that makes this OK?

Reply:
The ASIO-based client allows programmatic addition of certs. For the WinHTTP-based client, certs have to be added to the Windows cert store, which  C++ REST SDK doesn't provide any means to do. It has to be done via alternative means. I guess if the necessary cert is installed on the test system, then the call to disable validation can just be removed.

Reply:
Thinking about it, the best thing to do right now is to keep the `#if` which will fix the build but remove the `#else` that disables the cert validation, which will mean the test fails in this weird build config.

Would that be acceptable, @BillyONeal?

Adding C++ REST SDK support for working with [a temporary cert store for WinHTTP](https://stackoverflow.com/questions/1557710/winhttp-how-to-use-a-temporary-certificate-store) could be on somebody's to do list. Wrapping that entirely new functionality and the ASIO backend-specific mechanism via `set_ssl_context_callback` behind the same façade would then be next on the list.